### PR TITLE
Fix EZP-24380: Move section form management from PlatformUI to RepositoryForms

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-23480

### CODE REVIEW

**Comment by lolautruche** [comment]
Now ready for review

@andrerom @bdunogier @dpobel @yannickroger @glye @pspanja

**Comment by andrerom** [comment]
+1, before you rebase you can now merge in #307 if the change is ok for you, I rebased it on top of this one.

File: Resources/views/Section/view.html.twig [hunk 1]
```diff
@@ -38,28 +38,33 @@
...
-                            >
-                        {{ 'section.edit'|trans }}
-                    </a>
+            {% if canEdit %}
+                <a href="{{ path('admin_sectionedit', {'sectionId': section.id}) }}"
+                   class="pure-button ez-button" data-icon="&#xe606;">
+                    {{ 'section.edit'|trans }}
+                </a>
 
+                <a href="{{ path("admin_sectiondelete", {"sectionId": section.id}) }}"
```
Comment (dpobel) [inline Resources/views/Section/view.html.twig]: that should be button to post a form, deleting something with a GET request is bad.

File: Controller/SectionController.php [hunk 2]
```diff
@@ -176,63 +101,40 @@ public function viewAction($sectionId)
...
-                        ['sectionId' => $newSection->id]
-                    )
-                );
-            } catch (UnauthorizedException $e) {
-                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
-            } catch (InvalidArgumentException $e) {
-                $this->addAlreadyExistErrorMessage();
-            }
+        try {
+            $section = $this->sectionService->createSection(new SectionCreateStruct([
```
Comment (dpobel) [inline Controller/SectionController.php]: by doing that, as soon as the user clicks on "Create a new section", the section is created before he validates the form. Creating something on the server with a GET request is bad (like deleting) and worse, if the user navigates elsewhere or closes his browser, the new section has been created but the uses has absolutely no clue about that.

Comment (lolautruche) [inline Controller/SectionController.php]: I know, just like for ContentType... However this is needed to workaround the fact that we have different structs in the API for create and for update.

Comment (lolautruche) [inline Controller/SectionController.php]: Alternative would be to implement versioning for sections like @andrerom already suggested, but looks very overkill to me.

File: Resources/views/Section/list.html.twig [hunk 3]
```diff
@@ -49,35 +47,59 @@
...
+                        <td>
+                        {% if canEdit %}
+                            <a href="{{ path('admin_sectionedit', {'sectionId': section.id}) }}" class="pure-button ez-button" data-icon="&#xe606;">{{ 'section.edit'|trans }}</a>
+                        {% else %}
+                            <span class="pure-button ez-button pure-button-disabled" data-icon="&#xe606;">{{ 'section.edit'|trans }}</span>
+                        {% endif %}
+                        </td>
+                        <td>
+                        {% if canEdit %}
+                            <a href="{{ path('admin_sectiondelete', {'sectionId': section.id}) }}"
```
Comment (dpobel) [inline Resources/views/Section/list.html.twig]: button + form here as well

File: Resources/views/Section/view.html.twig (hunk 1 above)
Comment (lolautruche) [inline Resources/views/Section/view.html.twig]: And doing this by posting a form complicates things a lot :smiley:

Comment (bdunogier) [inline Resources/views/Section/view.html.twig]: Unfortunately I have to agree on this. A pragmatic reason is that doing it via GET would skip the Form Token. And that's kinda very bad.

**Comment by lolautruche** [comment]
All remarks taken into account. Ready for another round of review, especially from master @dpobel :smile:

**Comment by dpobel** [comment]
+1

Yoda
The real master is saying: *Merge this pull request, you can* :)

Joke aside, we'll have to change the redirection after the form submission as described in https://jira.ez.no/browse/EZP-23700 so to avoid having the inconsistency between the app URI and what is actually displayed.

**Pull request merged**